tests/filecheck/transforms/convert_memref_args_to_ptr.mlir

```diff
@@ -0,0 +1,42 @@
+// RUN: xdsl-opt -p convert-memref-to-ptr{lower_func=true} --split-input-file --verify-diagnostics %s | filecheck %s
+
+// CHECK:       builtin.module {
+
+// CHECK-NEXT:    func.func @declaration(!ptr_xdsl.ptr) -> ()
+func.func @declaration(%arg : memref<2x2xf32>)
+
+
+// CHECK-NEXT:    func.func @simple(%arg : !ptr_xdsl.ptr) {
+// CHECK-NEXT:      func.return
+// CHECK-NEXT:    }
```

Comment:
Another wee nit: usually we add the expected matching IR below each test case.

Reply:
So we're not super consistent but I've been going with the MLIR-style check above recently, and recommended it here.